### PR TITLE
[fix](load) fix load rpc timeout

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

```
Exception:
com.mysql.cj.jdbc.exceptions.CommunicationsException: Communications link failure
The last packet successfully received from the server was 1,800,102 milliseconds ago. The last packet sent successfully to the server was 1,800,105 milliseconds ago.
```

Not merge to 2.0 branch

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Review by Gabriel39 — DISMISSED** [review]
LGTM

**Review by HappenLee — DISMISSED** [review]
LGTM

File: be/src/vec/sink/writer/vtablet_writer.cpp [hunk 1]
```diff
@@ -996,28 +995,30 @@ void VTabletWriter::_send_batch_process() {
...
             });
+            opened_nodes += index_channel->num_node_channels();
         }
 
-        // if there is no channel, maybe auto partition table. so check does there have had running channels ever.
-        if (running_channels_num == 0 && had_effect) {
+        // auto partition table may have no node channel temporarily. wait to open.
+        // for auto partition tables, there's a situation: we haven't open any node channel but decide to cancel the task.
+        //  then the judge will never be true because opened_nodes won't increase. so we have to specially check wether we called close.
+        if ((opened_nodes != 0 && running_channels_num == 0) || _try_close) {
```
Comment (HappenLee) [inline be/src/vec/sink/writer/vtablet_writer.cpp:1017]: review the `_try_close` the logic

**Comment by zclllyybb** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
run buildall

**Review by BiteTheDDDDt — APPROVED** [review]

**Review by Gabriel39 — APPROVED** [review]
LGTM

File: be/src/vec/sink/writer/vtablet_writer.cpp [hunk 2]
```diff
@@ -17,6 +17,7 @@
 
 #include "vtablet_writer.h"
 
+#include <bits/ranges_algo.h>
```
Comment (adonis0147) [inline be/src/vec/sink/writer/vtablet_writer.cpp:20]: The header `bits/ranges_algo.h` isn't portable.
```suggestion
#include <ranges>
```

Comment (zclllyybb) [reply be/src/vec/sink/writer/vtablet_writer.cpp:20]: yes. will fix immediately

**Pull request merged**